shnet_main.c

```diff
@@ -396,12 +382,50 @@ ssize_t shnet_port_read(struct file *file, char __user *buf, size_t size,
 	return sz;
 }
 
+ssize_t shnet_port_write(struct file *file, const char __user *buf, size_t size,
+			 loff_t *ppos)
+{
+	int rc, sz = 0;
+	struct shnet_req req;
+
+	while (1) {
+		if (sz + sizeof(struct shnet_req) > size)
+			goto out;
+
+		rc = copy_from_user(&req,
+				    (struct shnet_req __user *)(buf + sz),
+				    sizeof(req));
+		if (rc) {
+			pr_err("Fail to copy from user buf, pos=%d\n", sz);
+			return sz;
+		}
+
+		if ((req.flags & SHNET_REQ_SIGNATURE) != SHNET_REQ_SIGNATURE) {
```

Comment:
Regarding the use of signature, I am not against it, but we do need to check is the "standard" way to do this kind of call